crates/types/src/lib.rs

```diff
@@ -46,17 +48,21 @@ pub struct ValidatorConfig<KEY: SignatureKey> {
     pub private_key: KEY::PrivateKey,
     /// The validator's stake
     pub stake_value: u64,
+    /// the validator's key pairs for state signing/verification
+    pub state_key_pairs: StateSigKeyPairs,
 }
 
 impl<KEY: SignatureKey> ValidatorConfig<KEY> {
     /// generate validator config from input seed, index and stake value
     #[must_use]
     pub fn generated_from_seed_indexed(seed: [u8; 32], index: u64, stake_value: u64) -> Self {
         let (public_key, private_key) = KEY::generated_from_seed_indexed(seed, index);
+        let state_key_pairs = generat_key_pair_from_seed_indexed(seed, index);
```

Comment:
Are we using the same signature key scheme for `state_key_pairs`? I'm a little bit confused on these three generation functions. I know  `generated_from_seed_indexed` is for key pair generation inside `hotshot`, and `KEY::generated_from_seed_indexed` is calling key pair generation with given seed inside `jellyfish`, what's the third one `generat_key_pair_from_seed_indexed` used for? Do we want to give it a different naming?

Reply:
Here I'm just copying things from `hotshot-signature-keys`. In Jellyfish, it's always taking some `rng` as an input. This function just wraps around generating `rng` from some seed.

Reply:
Oh sorry I forgot the previous two are all for generation in `hotshot`, but one for ValidatorConfig and the other for general usage by calling the one inside jellyfish. I didn't find this function: `generat_key_pair_from_seed_indexed()`, just wondering is it the same as calling `KEY::generated_from_seed_indexed()`?

Reply:
Just noticed it's a typo here.
https://github.com/EspressoSystems/HotShot/pull/2123/files#diff-65484181673c3c8c0a2c4b52a749c666a9028434c97f571ccd5b34d758eb2788R62-R74

Reply:
That makes sense, thanks!